int32[] deltaBlock
int32[] rasterizedRoom
float32 visionScope
---
int32[] sampledRoom